ocpmodels/trainers/base_trainer.py

```diff
@@ -728,9 +728,14 @@ def save_results(self, predictions, results_file, keys):
             # Because of how distributed sampler works, some system ids
             # might be repeated to make no. of samples even across GPUs.
             _, idx = np.unique(gather_results["ids"], return_index=True)
-            gather_results["ids"] = gather_results["ids"][idx]
+            gather_results["ids"] = np.array(gather_results["ids"])[idx]
             for k in keys:
-                gather_results[k] = gather_results[k][idx]
+                if k == "forces":
+                    gather_results[k] = np.array(
+                        gather_results[k], dtype=object
+                    )[idx]
```

Comment:
@abhshkdz @anuroopsriram If we have a better way for doing this let me know, this works for the time being but looks a bit hacky.